packages/vibrant-components/src/lib/ResourceList/ResourceItemProps.ts

```diff
@@ -13,13 +13,19 @@ export type ResourceItemProps = {
   ActionComponent?: ReactElement<ComponentProps<typeof IconButton>>;
 
   width?: ResponsiveValue<number | `${number}%`>;
-  id?: string;
-  onToggleSelect?: (args: { id?: string; selected: boolean }) => void;
-
   selectable?: boolean;
   defaultValue?: boolean;
   selected?: boolean;
-};
+} & (
+  | {
+      id: string;
+      onToggleSelect?: (args: { id: string; selected: boolean }) => void;
+    }
+  | {
+      id?: never;
```

Comment:
이 경우 id: undefined; 와 같아서 아래와 같이 수정하면 좋을 듯 합니다!

```suggestion
      id: never;
```